### PR TITLE
Remove X360/PS3 code

### LINKED ISSUE #512: Remove controller support
It just is not feasible to use controllers for these game types, so remove all related code to using controllers as input.

- [ ] Remove in_joystick related code, files, commands, etc
- [ ] Remove controller settings in the base Options -> Mouse tab
- [ ] Remove all references to "Rumble" or vibration

### LINKED ISSUE #519: Remove all Xbox 360 Related Files/Code
As we are not an Xbox game, nor do we have an Xbox devkit, we can safely remove all references to the code added for Xbox 360 support. We will need engine to completely rid it, but we can remove a lot as is from our game files.

Things to look for:

- `IsX360()`
- `#ifdef _X360`
- string search for "xbox" or "360"

### PR DESCRIPTION
**Pull request by xen-000**
Closes #519

This PR aims at removing as much as possible of the code that was implemented for Xbox 360 or PS3 support as there is no reason to keep it. Some of the remaining code will be removed as part of #512.

### Checklist
- [x] If there is a localization token change, I have updated the `momentum_english_ref.res` file with the changes, ran `tokenizer.py` to generate an up-to-date localization file, and have committed both the `.res` file changes and the new localization `.txt` file
- [x] If I introduced new h/cpp files, I have added them to the appropriate project's VPC file (`server_momentum.vpc` / `client_momentum.vpc` / etc)
- [x] My commits are relatively small and scoped to the best of my ability
- [x] My branch has a clear history of changes that can be easy to follow when being reviewed commit-by-commit
- [x] My branch is functionally complete; the only changes to be done will be those potentially requested in code review

### CODE REVIEW

File: mp/src/public/tier0/platform.h [hunk 1]
```diff
@@ -225,16 +178,7 @@ typedef signed char int8;
...
-	#if !defined( _GAMECONSOLE )
-		#define _GAMECONSOLE
-	#endif
-	#define IsPC()			false
-	#define IsGameConsole()	true
-#else
-	#define IsPC()			true
-	#define IsGameConsole()	false
-#endif
+#define IsPC()			true
```
Comment (Gocnak) [inline mp/src/public/tier0/platform.h]: It could be argued that this isn't necessary anymore either.

File: mp/src/public/tier1/iconvar.h [hunk 2]
```diff
@@ -67,7 +67,6 @@ class CCommand;
 
 #define FCVAR_NOT_CONNECTED		(1<<22)	// cvar cannot be changed by a client that is connected to a server
 #define FCVAR_MATERIAL_SYSTEM_THREAD (1<<23)	// Indicates this cvar is read from the material system thread
-#define FCVAR_ARCHIVE_XBOX		(1<<24) // cvar written to config.cfg on the Xbox
```
Comment (Gocnak) [inline mp/src/public/tier1/iconvar.h]: I would add this to the comments below with `// #define FCVAR_AVAILABLE (1<<24)`

File: mp/src/public/tier0/platform.h (hunk 1 above)
Comment (xen-000) [reply mp/src/public/tier0/platform.h]: That's true, it was used in one last place. Will remove.

File: mp/src/public/tier1/iconvar.h (hunk 2 above)
Comment (xen-000) [reply mp/src/public/tier1/iconvar.h]: Somehow didn't notice those comments, Will do.

**Pull request merged**